pkg/controller/infinispan/resources/config/config_controller.go

```diff
@@ -0,0 +1,170 @@
+}
+
+func (r reconcileConfig) InfinispanInstance(infinispan *ispnv1.Infinispan, ctrl *resources.Controller, kube *kube.Kubernetes, log logr.Logger) resources.Resource {
+	return &configResource{
+		infinispan: infinispan,
+		client:     r.Client,
+		scheme:     ctrl.Scheme,
+		kube:       kube,
+		log:        log,
+	}
+}
+
+func (r reconcileConfig) Types() []*resources.ReconcileType {
+	return []*resources.ReconcileType{{&corev1.ConfigMap{}, corev1.SchemeGroupVersion, true}}
+}
+
+func (r reconcileConfig) EventsPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return false
+		},
+	}
+}
+
+func Add(mgr manager.Manager) error {
+	return resources.CreateController(ControllerName, &reconcileConfig{mgr.GetClient()}, mgr)
+}
+
+func (c *configResource) Process() (reconcile.Result, error) {
+	xsite := &config.XSite{}
+	if c.infinispan.HasSites() {
+		// Check x-site configuration first.
+		// Must be done before creating any Infinispan resources,
+		// because remote site host:port combinations need to be injected into Infinispan.
+
+		// For cross site, reconcile must come before compute, because
+		// we need xsite service details to compute xsite struct
+		siteService := &corev1.Service{}
+		if result, err := infinispan.LookupResource(c.infinispan.GetSiteServiceName(), c.infinispan.Namespace, siteService, c.client, c.log); result != nil {
+			return *result, err
+		}
+
+		err := ComputeXSite(c.infinispan, c.kube, siteService, c.log, xsite)
+		if err != nil {
+			c.log.Error(err, "Error in computeXSite configuration")
+			return reconcile.Result{RequeueAfter: consts.DefaultWaitOnCreateResource}, nil
+		}
+	}
+
+	// TODO Remove after merging #675
+	c.infinispan.ApplyEndpointEncryptionSettings(c.kube.GetServingCertsMode(), c.log)
+
+	configMap, err := c.computeConfigMap(xsite)
+	if err != nil {
+		c.log.Error(err, "Could not create Infinispan configuration")
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	err = c.reconcileConfigMap(configMap)
+	if err != nil {
+		c.log.Error(err, "Error in reconcileConfigMap")
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (c configResource) computeConfigMap(xsite *config.XSite) (*corev1.ConfigMap, error) {
+	name := c.infinispan.Name
+	namespace := c.infinispan.Namespace
+
+	loggingCategories := c.infinispan.GetLogCategoriesForConfigMap()
+	config := config.CreateInfinispanConfiguration(name, loggingCategories, namespace, xsite)
+	// Explicitly set the number of lock owners in order for zero-capacity nodes to be able to utilise clustered locks
+	config.Infinispan.Locks.Owners = c.infinispan.Spec.Replicas
+
+	err := infinispan.ConfigureServerEncryption(c.infinispan, &config, c.client)
+	if err != nil {
+		return nil, err
+	}
+	configYaml, err := config.Yaml()
+	if err != nil {
+		return nil, err
+	}
+	lsConfigMap := infinispan.LabelsResource(name, "infinispan-configmap-configuration")
```

Comment:
We can make "infinispan-configmap-configuration"" a constant.

Reply:
Let's keep "as is" and plan to review all constants usage in separate submitted issue #573